common/src/Model/EntityNode.cpp

```diff
@@ -195,19 +193,14 @@ namespace TrenchBroom {
             return new EntitySnapshot(this);
         }
 
-        class CanAddChildToEntity : public ConstNodeVisitor, public NodeQuery<bool> {
-        private:
-            void doVisit(const WorldNode*)  override   { setResult(false); }
-            void doVisit(const LayerNode*)  override   { setResult(false); }
-            void doVisit(const GroupNode*)  override   { setResult(true); }
-            void doVisit(const EntityNode*) override { setResult(true); }
-            void doVisit(const BrushNode*)  override   { setResult(true); }
-        };
-
         bool EntityNode::doCanAddChild(const Node* child) const {
-            CanAddChildToEntity visitor;
-            child->accept(visitor);
-            return visitor.result();
+            return child->acceptLambda(kdl::overload {
+                [] (const WorldNode*)  { return false; },
+                [] (const LayerNode*)  { return false; },
+                [] (const GroupNode*)  { return false; },
```

Comment:
GroupNode and EntityNode should return true.

Reply:
In the original implementation, it did, but that was wrong. Entities can only have brushes as children. But I should split this commit in two.

Reply:
Ok, I have split the commit. Sorry for the confusion.